airgun/browser.py

```diff
@@ -211,6 +214,13 @@ def _get_selenium_browser(self):
                     ('chrome', 'firefox', 'ie', 'edge', 'phantomjs')
                 )
             )
+        if self._session:
+            # webdriver doesn't allow to add cookies unless we land on the target domain
+            # let's navigate to its invalid page to get it loaded ASAP
+            self._webdriver.get('https://{0}/404'.format(settings.satellite.hostname))
+            self._webdriver.add_cookie(
+                {'name': '_session_id', 'value': self._session.cookies.get_dict()['_session_id']}
+            )
```

Comment:
Can you extract that into own method, something like `_add_session_cookie`?
Then instead of copy-pasting this entire block for all `_get_<provider>_browser` implementations, you could just call this new method.